src/App.tsx

```diff
@@ -1,26 +1,118 @@
-/* eslint-disable max-len */
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID } from './api/todos';
+import { client } from './utils/fetchClient';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { Filter } from './types/Filter';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import { ErrorMessage } from './components/EroorMessage/ErrorMessage';
+import { TodoInput } from './components/TodoInput/TodoInput';
+import { Context } from './components/Context/Context';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const timeoutRef = useRef<NodeJS.Timeout | null>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [todosToDelete, setTodosToDelete] = useState<number[]>([]);
+
+  const getFilteredTodos = () => {
```

Comment:
it's better to use memo in such cases:

```suggestion
  const filteredTodos = useMemo(() => {
```
and add dependencies array below as second argument of useMemo